File: app/src/main/java/com/ignis/android_cleanarchitecture/data/entity/ForecastEntity.java
package com.ignis.android_cleanarchitecture.data.entity;

import io.realm.RealmObject;

/**
 * Forecasts Entity
 * <p>
 * Created by tamura_k on 2016/06/03.
 */
public class ForecastEntity extends RealmObject {

    //予報日
    private String date;
    //予報日(今日、明日、明後日のいずれか)
    private String dateLabel;
    //天気（晴れ、曇り、雨など）
    private String telop;
    //画像
    private ImageEntity image;
    //気温
    private TemperatureEntity temperature;

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getDateLabel() {
        return dateLabel;
    }

    public void setDateLabel(String dateLabel) {
        this.dateLabel = dateLabel;
    }

    public String getTelop() {
        return telop;
    }

    public void setTelop(String telop) {
        this.telop = telop;
    }

    public ImageEntity getImage() {
        return image;
    }

    public void setImage(ImageEntity image) {
        this.image = image;
    }

    public TemperatureEntity getTemperature() {
        return temperature;
    }

    public void setTemperature(TemperatureEntity temperature) {
        this.temperature = temperature;
    }

}

File: app/src/main/java/com/kazakago/cleanarchitecture/data/entity/weather/ForecastEntity.kt
package com.kazakago.cleanarchitecture.data.entity.weather

import android.arch.persistence.room.Embedded
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

@Entity
class ForecastEntity(
        @PrimaryKey(autoGenerate = true)
        val forecastId: Int,

        //予報日
        val date: String?,
        //予報日(今日、明日、明後日のいずれか)
        val dateLabel: String?,
        //天気（晴れ、曇り、雨など）
        val telop: String?,
        //画像
        @Embedded
        val image: ImageEntity?,
        //気温
        @Embedded
        val temperature: TemperatureEntity?
)